### PR TITLE
Github api: update to newest module version

### PR DESCRIPTION
**Pull request by danielbeardsley**
We were on an ancient version of the github api npm module, bump it to a more recent one.

### CODE REVIEW

File: bin/get-github-oauth-token [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      prompt: "Github Password:"
+   }, function(error, response) {
+      auth.password = response;
+      getOTP(auth);
+   })
+}
+
+function getOTP(auth) {
+   read({
+      prompt: "2-Factor Auth Code (leave blank for none):"
```
Comment (davidrans) [inline bin/get-github-oauth-token]: How does the user get their "2-Factor Auth Code"? Wouldn't we need to call `github.authenticate()` before they get a text?

**Comment by davidrans** [comment]
CR ☕  but I don't understand [how this works](https://github.com/danielbeardsley/cimpler/pull/71#discussion_r81194482).

File: bin/get-github-oauth-token [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env node
+
+var Github = require('github'),
+    read   = require('read');
+
+read({
```
Comment (MooreGuy) [inline bin/get-github-oauth-token]: should this be in a function called `authenticate`?

File: bin/get-github-oauth-token [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+   if (auth.otp) {
+      authorization.headers = {
+         "X-GitHub-OTP": auth.otp
+      };
+   }
+
+   var github = new Github();
+   auth.type = 'basic';
+   github.authenticate(auth);
```
Comment (MooreGuy) [inline bin/get-github-oauth-token]: Looks like this method doesn't exist, might be related to the comment above.

File: bin/get-github-oauth-token (hunk 1 above)
Comment (davidrans) [inline bin/get-github-oauth-token]: Hmm, maybe we should just be checking via the API if they have 2-Factor Auth turned on? https://developer.github.com/v3/auth/

Comment (danielbeardsley) [inline bin/get-github-oauth-token]: I forgot some people use SMS for 2FA... This worked fine with `Google Authenticator`

> For users with two-factor authentication enabled, Basic Authentication requires an extra step. When you attempt to authenticate with Basic Authentication, the server will respond with a 401 and an X-GitHub-OTP: required; :2fa-type header. This indicates that a two-factor authentication code is needed (in addition to the username and password). The :2fa-type in this header indicates whether the account receives its two-factor authentication codes via SMS or via an application.

That doesn't say if they actually *send* the sms when you make that failed api call. Maybe this whole script should just be one echo statement:

> Go to https://github.com/settings/tokens and create a token with `repo:status` permission

Which is a bit simpler.

File: bin/get-github-oauth-token (hunk 2 above)
Comment (danielbeardsley) [inline bin/get-github-oauth-token]: No. We ask for a username for info first, *then* send it to GH.

File: bin/get-github-oauth-token (hunk 3 above)
Comment (danielbeardsley) [inline bin/get-github-oauth-token]: ? [It does exist](https://github.com/mikedeboer/node-github#authentication)

**Comment by danielbeardsley** [comment]
QA 👍  (for non-sms 2FA)

File: bin/get-github-oauth-token (hunk 3 above)
Comment (MooreGuy) [inline bin/get-github-oauth-token]: Okay

**Comment by MooreGuy** [comment]
CR :+1:

File: bin/get-github-oauth-token (hunk 1 above)
Comment (davidrans) [inline bin/get-github-oauth-token]: Yeah, maybe the comment in the config should be:
```
Get an oauth token using bin/get-github-oauth-token or go to https://github.com/settings/tokens and create a token with repo:status permission

Comment (danielbeardsley) [inline bin/get-github-oauth-token]: Done!

**Comment by davidrans** [comment]
CR ☕

**Comment by davidrans** [comment]
CR ☕️

**Comment by danielbeardsley** [comment]
I've updated the pull desc to reflect the actual changes.

**Comment by MooreGuy** [comment]
CR 👍

File: plugins/cli.js [hunk 4]
```diff
@@ -1,5 +1,4 @@
-var util       = require('util'),
-    _          = require('underscore');
+var _          = require('underscore');
```
Comment (MooreGuy) [inline plugins/cli.js]: So much spaaaceee

**Comment by MooreGuy** [comment]
CR 👍

**Pull request merged**